pkg/operator/controller/ingress/nodeport_service.go

```diff
@@ -145,13 +164,38 @@ func nodePortServiceChanged(current, expected *corev1.Service) (bool, *corev1.Se
 		cmp.Comparer(cmpServiceAffinity),
 		cmpopts.EquateEmpty(),
 	}
-	if cmp.Equal(current.Spec, expected.Spec, serviceCmpOpts...) {
+	if !cmp.Equal(current.Spec, expected.Spec, serviceCmpOpts...) {
+		changed = true
+	}
+
+	annotationCmpOpts := []cmp.Option{
+		cmpopts.IgnoreMapEntries(func(k, _ string) bool {
+			return !managedNodePortServiceAnnotations.Has(k)
+		}),
+	}
+	if !cmp.Equal(current.Annotations, expected.Annotations, annotationCmpOpts...) {
+		changed = true
+	}
+
+	if !changed {
 		return false, nil
 	}
 
 	updated := current.DeepCopy()
 	updated.Spec = expected.Spec
 
+	if updated.Annotations == nil {
+		updated.Annotations = map[string]string{}
+	}
+	for annotation := range managedNodePortServiceAnnotations {
+		currentVal, have := current.Annotations[annotation]
+		expectedVal, want := expected.Annotations[annotation]
+		if want && (!have || currentVal != expectedVal) {
+			updated.Annotations[annotation] = expected.Annotations[annotation]
+		} else if have && !want {
+			delete(updated.Annotations, annotation)
```

Comment:
Could you add a unit test for this scenario?

Reply:
Fixed.  However, I had to replace the test case for adding the annotation with test cases for updating and deleting.  To cover all three would make the test setup more complicated.  If you want coverage for all three, I'd just as soon add a separate test.  What do you think?

Reply:
I see. Current coverage looks good enough to me! Thanks.